### PR TITLE
etcd: add test

### PR DESCRIPTION
**Pull request by JasonBoyles**
This test starts an etcd process as a single-node cluster, stores a test
string in the key-value store, retrieves it, and compares it against the
original string. This tests whether etcd can start properly, select
itself as a leader, accept a PUT request, store the value, then return
it on a GET. We kill the etcd process once we are done.

### CODE REVIEW

**Comment by JasonBoyles** [comment]
This test works locally on my Yosemite machine. I'm wondering if Popen is behaving differently on Mavericks.

File: Library/Formula/etcd.rb [hunk 1]
```diff
@@ -20,6 +20,27 @@ def install
...
+      require "net/http"
+      test_string = "Hello from brew test!"
+      etcd_pid = fork do
+        exec "etcd", "--force-new-cluster", "--data-dir=#{testpath}"
+      end
+      # sleep to let etcd get its wits about it
+      sleep 10
+      etcd_uri = "http://127.0.0.1:4001/v2/keys/brew_test"
+      system "curl", "--silent", "-L", etcd_uri, "-XPUT", "-d", "value=#{test_string}"
+      curl_output = IO.popen(["curl", "--silent", "-L", etcd_uri]).read
```
Comment (mikemcquaid) [inline Library/Formula/etcd.rb]: Use `shell_output` instead of `popen` here.

**Comment by JasonBoyles** [comment]
Now using `shell_output` instead of `popen`

Thanks for the pointer @mikemcquaid!

**Comment by mikemcquaid** [comment]
Thanks @JasonBoyles!

**Pull request closed without merging**